### DISCUSSION TITLE
Trying to fetch with Vercel directs to wrong domain

### DISCUSSION THREAD

**Discussion by b0wdyy** [vercel/vercel#5611] [Help]
Whenever I try to make an api request to my other domain 'foo.bar/api/v1/users' it concatenates to my vercel url making it 'vercel-app.bar/foor.bar/api/v1/users'. Does anyone know a fix? 

![Screenshot 2020-12-28 at 21 08 30](https://user-images.githubusercontent.com/58066652/103240583-06c9d000-4951-11eb-8bd1-22730e9d331b.png)

**Comment by paulogdm** [comment]
It is hard to assess what went wrong without some code. Please share a repository because this is likely caused by a code-level problem.

**Comment by e-Dylan** [comment]
I faced the exact same issue in the same situtation you are in. I am running my front end React client through Vercel and fetching my backend AWS VPS server running express at /api/. For this, I am using environment variables to define my api url based on my NODE_ENV. 

TLDR; 
Remove the quotes in your environment variable values that define the API_URL, vercel adds quotes automatically so it ends up with double quotes in the .env

Vercel uses environment variables defined through their dashboard in deployment. When I defined my variables under Vercel's Project Settings -> Environment Variables tab, using plaintext var types, I set the values as "http:mydomain.com:1337/api" (1337 is the port my express app is running on).

Vercel takes the defined Value field (including my quotes, my fetch url needs to be a string) and puts it into its own string environment variable, therefore it ends up as:

API_URL=""http://mydomain.com:1337/api""

Notice the double quotes. When this occurs, it will by default append this url to the end of your front-end's url and thus fetch calls with return 404 errors.

For debugging, you can use the vercel CLI - "vercel env pull" will show you the .env file it is running with when deployed. You can then run the project using "vercel dev" and alter your variables until they work. Make sure once you find what works that you change your project's environment variables to your fixed ones at the Environment Variables dashboard in Vercel.
